client/api/eventsApi.js

```diff
@@ -25,12 +25,20 @@ export function addNewEvent (id, event) {
 }
 
 /**
- * @typedef {object} Event
- * @property {number} id
+ * Event object details
+ * @typedef {object<string, any>} Event
+ * @property {number} id - event id
+ * @property {string} name- Name of event
+ * @property {string} description - description of event
+ * @property {string} topic - Event category
+ * @property {number} startDate - start date of event
+ * @property {number} endDate - end date of event
  */
 /**
  * Request array of events from projects table
+ *
  * @returns {Promise.<Event[]>}
+ * @type {Event}
```

Comment:
❕ I'm not sure that `@type` would work here.  Generally `@type` is used to document the type of the symbol following the docblock (`showAllEvents`, in this case), so `@type {function}` would be more correct (but JSDoc is smart enough to figure this out without being told explicitly).

There's a vague note in [the JSDocs documentation](https://jsdoc.app/tags-type.html) that `@type` could be used to document a function's return type, but I'd never heard of it being used this way before (also it contradicts the existing `@returns {Promise.<Event[]>}`).